File: app/src/main/java/com/wangyang/tinner/Book.java
package com.wangyang.tinner;

import android.os.Parcel;
import android.os.Parcelable;

public class Book implements Parcelable {
    private String bookName;
    private String bookId;


    protected Book(Parcel in) {
        bookName = in.readString();
        bookId = in.readString();
    }

    public static final Creator<Book> CREATOR = new Creator<Book>() {
        @Override
        public Book createFromParcel(Parcel in) {
            return new Book(in);
        }

        @Override
        public Book[] newArray(int size) {
            return new Book[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(bookName);
        dest.writeString(bookId);
    }
}

File: app/src/main/java/com/wangyang/tinner/TEST/Book.kt
package com.wangyang.tinner.TEST

import android.os.Parcel
import android.os.Parcelable

class Book() : Parcelable{
    var bookId:Int? = null
    var bookName:String? = null

    constructor(parcel: Parcel) : this() {
        bookId = parcel.readValue(Int::class.java.classLoader) as? Int
        bookName = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(bookId)
        parcel.writeString(bookName)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Book> {
        override fun createFromParcel(parcel: Parcel): Book {
            return Book(parcel)
        }

        override fun newArray(size: Int): Array<Book?> {
            return arrayOfNulls(size)
        }
    }
}